service/src/main/java/bio/terra/workspace/service/resource/GcpFlightExceptionUtils.java

```diff
@@ -19,4 +21,18 @@ public static void handleGcpBadRequestException(GoogleJsonResponseException e)
       throw new BadRequestException(e.getDetails().getMessage());
     }
   }
+
+  /**
+   * Map GCP status codes to WSM exceptions. These are errors that we DO NOT want to retry in the
+   * flight.
+   */
+  public static void handleGcpNonRetryableException(GoogleJsonResponseException e) {
+    int statusCode = e.getStatusCode();
+    switch (HttpStatus.valueOf(statusCode)) {
```

Comment:
I'm not 100% sure about this list. I wanted parallel behavior to `pollAndRetry`.

For GCE instance we definitely want to bail on FORBIDDEN and we don't ever expect UNAUTHORIZED.

I _think_ we want to bail on TOO_MANY_REQUESTS, since retrying will be making more requests.

I _think_ we do not want to bail on CONFLICT. I'm not sure what it means in this case. If it means, "wait a minute, I'm busy on that thing" (like it does for WSM and Sam), we would want to retry.


Question